### PR TITLE
Feature auth support

### PR DESCRIPTION
**Pull request by 168668660**
module和component鉴权接口android和ios都已添加

### CODE REVIEW

**Comment by sospartan** [comment]
@zshshr @cxfeng1 pls review this PR.

@168668660  
1. It's 'Validate' not 'auth', change this PR's title.
2. Do not commit with chinese any longer, it's not meet our contribute requirement. Thanks. 😄

File: android/sdk/src/main/java/com/taobao/weex/WXSDKInstance.java [hunk 1]
```diff
@@ -294,6 +294,11 @@
   private boolean isCommit=false;
   private WXGlobalEventReceiver mGlobalEventReceiver=null;
   private boolean trackComponent;
+  private boolean needValidate = false;
```
Comment (zshshr) [inline android/sdk/src/main/java/com/taobao/weex/WXSDKInstance.java]: 成员变量命名用骆驼命名，以m开头

File: android/sdk/src/main/java/com/taobao/weex/WXSDKInstance.java [hunk 2]
```diff
@@ -559,6 +564,9 @@ private void renderByUrlInternal(String pageName,
     ensureRenderArchor();
     pageName = wrapPageName(pageName, url);
     mBundleUrl = url;
+    if(WXSDKManager.getInstance().getValidateProcessor()!=null) {
+      needValidate = WXSDKManager.getInstance().getValidateProcessor().needValidate(mBundleUrl);
+    }
```
Comment (zshshr) [inline android/sdk/src/main/java/com/taobao/weex/WXSDKInstance.java]: needValidate 函数名建议改成 requestPermission或 requestValidation

**Comment by sospartan** [comment]
@168668660 pls fix the conflicts.

@cxfeng1 pls review the iOS part.

**Comment by sospartan** [comment]
@168668660  what's going on?

**Pull request closed without merging**